### PR TITLE
Avoid compiler crash in varbit packet extract

### PR DESCRIPTION
**Pull request by usha1830**
Avoid compiler crash by handling possible types of length argument in packet extract for header containing varbit field

### CODE REVIEW

File: backends/dpdk/dpdkHelpers.cpp [hunk 1]
```diff
@@ -908,37 +908,46 @@ bool ConvertStatementToDpdk::preorder(const IR::MethodCallStatement *s) {
...
+                         }
+
+                        IR::ID tmpName(refmap->newName(baseName + "_extract_tmp"));
+                        BUG_CHECK(metadataStruct, "Metadata structure missing unexpectedly!");
+                        metadataStruct->fields.push_back(
+                                new IR::StructField(tmpName, length->expression->type));
+                        auto tmpMember = new IR::Member(new IR::PathExpression("m"), tmpName);
+                        add_instr(new IR::DpdkMovStatement(tmpMember, length->expression));
+                        add_instr(new IR::DpdkShrStatement(tmpMember, tmpMember,
+                                new IR::PathExpression("0x3")));
```
Comment (kamleshbhalui) [inline backends/dpdk/dpdkHelpers.cpp]: why this is path expression it should be constant right?

Comment (usha1830) [reply backends/dpdk/dpdkHelpers.cpp]: Yes. Right. I will update it to constant. This was existing code which I moved around, not sure why it was added as PathExpression.

**Review by kamleshbhalui — APPROVED** [review]

**Review by psivanup — APPROVED** [review]
Looks fine to me.

**Comment by usha1830** [comment]
Closing as upstreamed

**Pull request closed without merging**